### ISSUE TITLE
ST messing with entities in other mods which already have perfectly passable collision boxes

### ISSUE THREAD

**Issue by ghost** [Suprcheese/Squeak-Through#6]
Hi,

I've had to give up my leisure time to tweak one of my mods to undo unwanted changes created by your mod. The entity in question was a 2x2 lamp which already had an extremely generous collision box which was perfectly possible to walk around. Your mod reduced the collision box so much that the lamps could be placed in a way that made them overlap. This was completely unintended and unwanted behaviour and it was me that had to field the complaints and undo the damage. I shouldn't have to add conditional code to my mods to safeguard them against unwanted interference by third parties. Nor should I have to maintain an exception list for every mod I make and pressure you to include it in your code.

Squeak Through should stop messing with entities from other mods, period. Confine it to vanilla entities where the properties are well known. Other modders are perfectly capable of designing their own collision boxes how they want them.

I will be marking ST as "incompatible" in any future mod I publish until this issue is resolved.

**Comment by Danny-Alexander** [comment]
Hi, I'm Nommy on the mod portal (verification in last post).

_Suprcheese, I don't mind if you add me as a colab or not. I'm not sure if I'm going to be putting any time into this or not but I can submit pull requests for you to check out either way AFAIK. It might be worth splitting this into 3 issues if you feel like it, or perhaps I could figure out how._ 

Deadlock989, so to break this and [our discussion on the mod portal](https://mods.factorio.com/mod/Squeak%20Through/discussion/5cdf13957b286b000c5efdf8) down, you've 3 issues related to Squeak Through (please let me know if I've missed anything):


### 1. [Bug Report] 2x2 Entities with passable collision boxes are reduced below 2x2 size
It would help anyone looking into this to be provided with details on exactly how to reproduce it.

You're talking about the Deadlock's Larger Lamps version 1.2.4 I assume?
Both 2x2 lamps?

The fastest way to solve this I expect would be to submit a pull request to add functionality to determine if an entity is passable based on it's tile dimensions vs collision box dimensions as that would solve the problem without reducing functionality and require the least time from anyone maintaining this mod.


## 2. [Feature Request] Allow excluding entities from having their collision boxes modified
e.g
```
{
    collision_box_modifiable = false
}
```
I think this would be a good addition as it would not reduce the end user experience and provide another way to prevent incompatibility errors.

I'd suggest a mod setting something like:
```
Squeak through behaviour:
1. Shrink only entities marked as modifiable (default)
2. Shrink all entities not marked as unmodifiable
3. Shrink all entities regardless (incompatibility warning)
```
Pros
 - Does not reduce functionality
 - Provides method of incompatibility avoidance

Cons
 - Option 2 & 3 may still result in compatibility conflicts with 'unknown' mods


### 3. [Feature Request] Only modify mod added entities marked as modifiable, period
TLDR: My suggestion: No, it would break one of the major features of the mod.

My intention has always been to provide players with more diversity and control over what kind of experience they create to play with. I think many people like having a mod which allows you to walk over waist high pipes and between massive buildings etc as it adds to immersion and 'quality of life' and they would much rather it worked on ALL pipes, not just those from mods who's authors who knew of Squeak Through and enabled it, like you demand.

Sure, this creates the possibility for compatibility issues and there are many ways of dealing with those which don't reduce the variety of end user experiences.

There's existing exclusion code which it's simple to create pull requests for right now.

There could (and I would say should) be a method to exclude entities who's functionality is dependant on specific collision box sizes from being modified. (#2 above).

There's room for improvement in the algorithm used to determine what and how much to change collision boxes (I assume based on the information you've provided). 

Providing information to reproduce the error you had related to this and even code to improve the algorithm to detect that no modification was necessary in that case and prevent that error would be far more productive than complaining about how your mod was changed and writing work around to prevent the issue in your mod, not to mention it would benefit players who enjoy using either mod.

I just wanted to make sure you're aware that if you'd approached the problem you had differently I expect it would be solved just the same but in a manner which benefits everyone far more (all other players and modders) and all the complaining and and demanding etc does nothing to motivate me to work on this, just FYI.

Thanks for the report, info and input etc. it was helpful.

**Comment by Suprcheese** [comment]
Thanks to PR https://github.com/Suprcheese/Squeak-Through/pull/8 , this issue should be addressed via the implementation of a solution incorporating point 2 above, as well as an optional setting to enable behavior of point 3 above. These will be implemented as of v1.4.0 of Squeak Through, which I will push later today.

All credit for these additions is to the PR author [Foxite](https://github.com/Foxite).
